src/DDTrace/Util/CodeTracer.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+namespace DDTrace\Util;
+
+use DDTrace\GlobalTracer;
+
+class CodeTracer
+{
+    /**
+     * @var CodeTracer
+     */
+    private static $instance;
+
+    /**
+     * @return CodeTracer
+     */
+    public static function getInstance()
+    {
+        if (self::$instance === null) {
+            self::$instance = new CodeTracer();
```

Comment:
Just for funzies. :)
```suggestion
            self::$instance = new self();
```